### DISCUSSION TITLE
Dynamic chunk cache (Cache chunks only when the page is accessed)

### DISCUSSION THREAD

**Discussion by codefln** [vite-pwa/vite-plugin-pwa#727] [General] [answered]
I have a large application, and when I first access the page, I receive about 800 requests to load all the service worker chunks. How can I implement the idea so that when accessing the main page, only the chunks needed for the main page are loaded, and after navigation, the chunks are dynamically added?

**Comment by userquin** [comment, marked as answer]
that's the precaching logic, you cannot change that behavior, you can only disable precaching and use runtime caching with custom caches (use custom sw => injectManifest strategy)

https://developer.chrome.com/docs/workbox/modules/workbox-precaching
